syntax = "proto3";

package convenio;

option java_package = "br.com.clinica.convenio";
option java_multiple_files = true;
option java_outer_classname = "ConvenioProto";

message ValidarConvenioRequest {
  // Autenticação (para validarUsuario)
  string email = 1;
  string senha = 2;

  // Dados do Convênio e Paciente
  string numeroCarteirinha = 3;
  string dataDeValidade = 4;
  string cpf = 5;            // Necessário para o getPatientId
}

message ValidarConvenioResponse {
  bool sucesso = 1;
  string mensagem = 2;
}

service ConvenioService {
  rpc validarConvenio(ValidarConvenioRequest) returns (ValidarConvenioResponse);
}